ChallengesWithTestsMarkVII/ChallengesSet02.cs

```diff
@@ -1,58 +1,159 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace ChallengesWithTestsMarkVII
 {
     public class ChallengesSet02
     {
         public bool CharacterIsALetter(char c)
         {
-            throw new NotImplementedException();
+            if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z'))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
         }
 
         public bool CountOfElementsIsEven(string[] vals)
         {
-            throw new NotImplementedException();
+            if (vals.Length % 2 == 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
         }
 
         public bool IsNumberEven(int number)
         {
-            throw new NotImplementedException();
+            if (number % 2 == 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
         }
 
         public bool IsNumberOdd(int num)
         {
-            throw new NotImplementedException();
+            if (num % 2 != 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+                
+
         }
 
         public double SumOfMinAndMax(IEnumerable<double> numbers)
         {
-            throw new NotImplementedException();
+            if(numbers?.Any() != true)
+            {
+             return 0;
+            }
+
+            else
+            {
+                return numbers.Min() + numbers.Max();
+            }
+
+
         }
 
         public int GetLengthOfShortestString(string str1, string str2)
         {
-            throw new NotImplementedException();
+            int str1length = str1.Length;
+            int str2length = str2.Length;
+
+            if (str1length >= str2length)
+            {
+                return str2length;
+            }
+            else if (str1length <= str2length)
+            {
+                return str1length;
+            }
+            else
+            {
+                return 0;
+            }
+            
+
+
+
         }
 
         public int Sum(int[] numbers)
         {
-            throw new NotImplementedException();
+            if (numbers == null)
+            {
+                return 0;
+            }
+            int sum = 0;
+            foreach (int i in numbers)
+                {
+                    sum += i;
+                }
+            return sum;
+
+
         }
 
         public int SumEvens(int[] numbers)
         {
-            throw new NotImplementedException();
+            if (numbers == null)
+            {
+                return 0;
+            }
+            int sum = 0;
+            foreach (int i in numbers)
+                if (i % 2 == 0)
```

Comment:
In my opinion, and it's only an opinion, I'd say choose one or the other: either don't use curly braces for single-line scope, or use them everywhere.
``` C#
foreach (int i in numbers)
    if (i % 2 =0)
        sum += i;
```
or
``` C#
foreach (int i in numbers)
{
    if (i % 2 =0)
    {
        sum += i;
    }
}
```